### PR TITLE
CAS-8310, add methods for retrieving spws and spw-average interval ma…

### PR DESCRIPTION
**Pull request by dmehring**
…ps for sub scans, update MSSummary to use these per sub scan values. Add unit tests to tMSMetaData.

### CODE REVIEW

File: ms/MSOper/MSMetaData.cc [hunk 1]
```diff
@@ -3516,15 +3524,35 @@ std::map<SubScanKey, MSMetaData::SubScanProperties> MSMetaData::_getSubScanPrope
...
+    std::map<SubScanKey, map<uInt, vector<Double> > >::const_iterator end = intervalSets.end();
+    Unit unit = intervalTimes->getFullUnit();
+    for (; iter!=end; ++iter) {
+        SubScanKey ssKey = iter->first;
+        map<uInt, vector<Double> > spwToIntervalSets = iter->second;
+        map<uInt, vector<Double> >::const_iterator siter = spwToIntervalSets.begin();
+        map<uInt, vector<Double> >::const_iterator send = spwToIntervalSets.end();
+        for (; siter!=send; ++siter) {
+            uInt spw = siter->first;
+            vector<Double> myIntervals = siter->second;
```
Comment (juliantaylor) [inline ms/MSOper/MSMetaData.cc]: this and `spwToIntervalSets` above seem to be unnecessary copies, the data appears to only to be read so references should do

**Comment by dmehring** [comment]
Updated as per your recommendation.

**Comment by dmehring** [comment]
@juliantaylor pinging for review.

**Comment by juliantaylor** [comment]
thanks

**Pull request merged**